src/Main.php

```diff
@@ -7,27 +7,24 @@
 use pocketmine\utils\TextFormat;
 
 class Main extends PluginBase {
-    public function onLoad() :void{
+	private string $version = "1.0.6";
+
+	public function onLoad() :void{
         $this->saveResource("config.yml");
         $config = $this->getConfig();
         $log = $this->getLogger();
-        $version = "1.0.5";
-        if ($config->get("config-version") == $version){
-            return;
-        } else {
-            $log->notice("Your config is outdated!");
-            $log->info("Your old config.yml was as old-config.yml");
-            @rename($this->getDataFolder(). 'config.yml', 'old-config.yml');
-            $this->saveResource("config.yml");
-        }
+        if ($config->get("config-version") == $this->version) return;
```

Comment:
Using that instead will require the user to update their config every-time the plug-in updates even if there's no changes to the config

Reply:
> Using that instead will require the user to update their config every-time the plug-in updates even if there's no changes to the config

```suggestion
        if ($config->get("config-version") === self::VERSION) return;
```
Maybe just make version a constant instead of property.